Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -0,0 +1,1203 @@
+        /// </summary>
+        private NetCollection _excludedSubnets;
+
+        /// <summary>
+        /// List of interface addresses to bind the WS.
+        /// </summary>
+        private NetCollection _bindAddresses;
+
+        /// <summary>
+        /// List of interface addresses to exclude from bind.
+        /// </summary>
+        private NetCollection _bindExclusions;
+
+        /// <summary>
+        /// Caches list of all internal filtered interface addresses and masks.
+        /// </summary>
+        private NetCollection _internalInterfaces;
+
+        /// <summary>
+        /// Flag set when no custom LAN has been defined in the config.
+        /// </summary>
+        private bool _usingPrivateAddresses;
+
+        /// <summary>
+        /// True if this object is disposed.
+        /// </summary>
+        private bool _disposed;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NetworkManager"/> class.
+        /// </summary>
+        /// <param name="configurationManager">IServerConfigurationManager instance.</param>
+        /// <param name="logger">Logger to use for messages.</param>
+#pragma warning disable CS8618 // Non-nullable field is uninitialized. : Values are set in InitialiseLAN function. Compiler doesn't yet recognise this.
+        public NetworkManager(IConfigurationManager configurationManager, ILogger<NetworkManager> logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _configurationManager = configurationManager ?? throw new ArgumentNullException(nameof(configurationManager));
+
+            _interfaceAddresses = new NetCollection(unique: false);
+            _macAddresses = new List<PhysicalAddress>();
+            _interfaceNames = new SortedList<string, int>();
+            _overrideUrls = new Dictionary<IPNetAddress, string>();
+
+            UpdateSettings((ServerConfiguration)_configurationManager.CommonConfiguration);
+            if (!IsIP6Enabled && !IsIP4Enabled)
+            {
+                throw new ApplicationException("IPv4 and IPv6 cannot both be disabled.");
+            }
+
+            NetworkChange.NetworkAddressChanged += OnNetworkAddressChanged;
+            NetworkChange.NetworkAvailabilityChanged += OnNetworkAvailabilityChanged;
+
+            _configurationManager.ConfigurationUpdated += ConfigurationUpdated;
+
+            Instance = this;
+        }
+#pragma warning restore CS8618 // Non-nullable field is uninitialized.
+
+        /// <summary>
+        /// Event triggered on network changes.
+        /// </summary>
+        public event EventHandler? NetworkChanged;
+
+        /// <summary>
+        /// Gets the singleton of this object.
+        /// </summary>
+        public static NetworkManager Instance
+        {
+            get => GetInstance();
+
+            internal set
+            {
+                _instance = value;
+            }
```

Comment:
```suggestion
            internal set => _instance = value;
```

Reply:
removed static instance from this version.